pennylane/labs/resource_estimation/resources_base.py

```diff
@@ -0,0 +1,384 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Base class for storing resources."""
+from __future__ import annotations
+
+import copy
+from collections import defaultdict
+from decimal import Decimal
+
+from pennylane.labs.resource_estimation.qubit_manager import QubitManager
+
+
+class Resources:
+    r"""A container to track and update the resources used throughout a quantum circuit.
+    The resources tracked include number of gates, number of wires, and gate types.
+
+    Args:
+        qubit_manager (QubitManager): A qubit tracker class which contains the number of available
+            work wires, categorized as clean, dirty or algorithmic wires.
+        gate_types (dict): A dictionary storing operations (ResourceOperator) as keys and the number
+            of times they are used in the circuit (int) as values.
+
+    **Example**
+
+    The resources can be accessed as class attributes. Additionally, the :class:`~.Resources`
+    instance can be nicely displayed in the console.
+
+    >>> h = re.resource_rep(re.ResourceHadamard)
+    >>> x = re.resource_rep(re.ResourceX)
+    >>> y = re.resource_rep(re.ResourceY)
+    >>>
+    >>> qm = re.QubitManager(work_wires=3)
+    >>> gt = defaultdict(int, {h: 10, x:7, y:3})
+    >>>
+    >>> res = re.Resources(qubit_manager=qm, gate_types=gt)
+    >>> print(res)
+    --- Resources: ---
+    Total qubits: 3
+    Total gates : 20
+    Qubit breakdown:
+    clean qubits: 3, dirty qubits: 0, algorithmic qubits: 0
+    Gate breakdown:
+    {'Hadamard': 10, 'X': 7, 'Y': 3}
+
+    .. details::
+        :title: Usage Details
+
+        The :class:`Resources` object supports arithmetic operations which allow for quick addition
+        and multiplication of resources.
+
+        .. code-block::
+
+            from collections import defaultdict
+
+            # Resource reps for each operator:
+            h = re.resource_rep(re.ResourceHadamard)
+            x = re.resource_rep(re.ResourceX)
+            z = re.resource_rep(re.ResourceZ)
+            cnot = re.resource_rep(re.ResourceCNOT)
+
+            # state of qubits:
+            qm1 = re.QubitManager(work_wires={"clean":2, "dirty":1})
+            qm1.algo_qubits = 3
```

Comment:
```suggestion
            qm1 = re.QubitManager(work_wires={"algo":3, "clean":2, "dirty":1})
```
Allowing one to do this might be cleaner.